test/system/session/test_special_values.py

```diff
@@ -0,0 +1,535 @@
+"""System tests for tables with special values (nulls, nans, infinities)."""
+
+# SPDX-License-Identifier: Apache-2.0
+# Copyright Tumult Labs 2025
+
+import datetime
+from typing import Dict, List, Optional, Tuple, Union
+
+import pandas as pd
+import pytest
+from numpy import sqrt
+from pyspark.sql import DataFrame
+from tmlt.core.utils.testing import Case, parametrize
+
+from tmlt.analytics import (
+    AddOneRow,
+    AddRowsWithID,
+    ColumnDescriptor,
+    ColumnType,
+    MaxRowsPerID,
+    ProtectedChange,
+    PureDPBudget,
+    Query,
+    QueryBuilder,
+    Session,
+)
+from tmlt.analytics._schema import Schema, analytics_to_spark_schema
+
+from ...conftest import assert_frame_equal_with_sort
+
+
+@pytest.fixture(name="sdf_special_values", scope="module")
+def null_setup(spark):
+    """Set up test data for sessions with special values."""
+    sdf_col_types = {
+        "string": ColumnDescriptor(ColumnType.VARCHAR, allow_null=True),
+        "int_no_null": ColumnDescriptor(ColumnType.INTEGER, allow_null=False),
+        "int_nulls": ColumnDescriptor(ColumnType.INTEGER, allow_null=True),
```

Comment:
Even if we're not having null/non-null versions of the other columns, I'd suggest using the `_nulls` suffix just to make everything completely explicit.